dipy/align/mattes.pyx

```diff
@@ -0,0 +1,1276 @@
+#!python
+#cython: boundscheck=False
+#cython: wraparound=False
+#cython: cdivision=True
+
+import numpy as np
+cimport numpy as cnp
+cimport cython
+from fused_types cimport floating
+
+from dipy.align.vector_fields cimport(_apply_affine_3d_x0,
+                                      _apply_affine_3d_x1,
+                                      _apply_affine_3d_x2,
+                                      _apply_affine_2d_x0,
+                                      _apply_affine_2d_x1)
+
+from dipy.align.transforms cimport (Transform)
+
+cdef extern from "dpy_math.h" nogil:
+    double cos(double)
+    double sin(double)
+    double log(double)
+
+class MattesBase(object):
+    def __init__(self, nbins):
+        r""" MattesBase
+        Base class for the Mattes' Mutual Information metric.
```

Comment:
I go with Strunk and White: http://www.bartleby.com/141/strunk.html

On Sun, Apr 19, 2015 at 9:22 AM, Omar Ocegueda <notifications@github.com>
wrote:

> In dipy/align/mattes.pyx
> <https://github.com/nipy/dipy/pull/628#discussion_r28654290>:
>
> > +                                      _apply_affine_3d_x1,
> > +                                      _apply_affine_3d_x2,
> > +                                      _apply_affine_2d_x0,
> > +                                      _apply_affine_2d_x1)
> > +
> > +from dipy.align.transforms cimport (Transform)
> > +
> > +cdef extern from "dpy_math.h" nogil:
> > +    double cos(double)
> > +    double sin(double)
> > +    double log(double)
> > +
> > +class MattesBase(object):
> > +    def __init__(self, nbins):
> > +        r""" MattesBase
> > +        Base class for the Mattes' Mutual Information metric.
>
> Fixed =) (removed the apostrophe and added missing reference)
>
> Just for the 'trivia', (don't put much attention to this, I have no
> preference about it =P ), apparently there's no rule for using apostrophe
> to indicate possession when the proper noun ends with 's', they just
> recommend to stay consistent:
> http://www.grammarbook.com/punctuation/apostro.asp
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/628/files#r28654290>.
>